extensions/vertx-core/deployment/src/main/java/io/quarkus/vertx/core/deployment/VertxOptionsConsumerBuildItem.java

```diff
@@ -0,0 +1,30 @@
+package io.quarkus.vertx.core.deployment;
+
+import java.util.function.Consumer;
+
+import io.quarkus.builder.item.MultiBuildItem;
+import io.vertx.core.VertxOptions;
+
+/**
+ * Provide a consumer of VertxOptions.
```

Comment:
Maybe add a note that the idea is to provide arbitrary customization of vertx?

Reply:
```
/**
 * Provide a consumer of VertxOptions to allow customization of
 * Vert.x system behavior, e.g. setting MetricsOptions to enable
 * and configure a metrics provider.
 * <p>
 * Consumers will be called in priority order (lowest to highest)
 * after VertxConfiguration has been read and applied.
 */
```